File: src/problem19/19.py
#!/usr/bin/python
import sys

def removeNeighboringDuplicateCharacters(input):
	characters = list(input)

	found = True
	while(found):
		found = False
		for i in range(0, len(characters) - 1):
			if characters[i] == characters[i+1]:
				del characters[i+1]
				del characters[i]
				found = True
				break
	
	if len(characters) == 0:
		return "Empty String"

	return ''.join(characters)


with open("19.out", 'w') as outFile:
	with open("19.in", 'r') as inFile:
		no_of_test_cases = int(next(inFile)) # read first line
		print no_of_test_cases

		for line in inFile: # read rest of lines
			result = removeNeighboringDuplicateCharacters(line.strip())
			outFile.write(result)
			outFile.write('\n')
			

File: src/problem19/19.js
/**
 *
 *
 * @export
 * @param {String} input
 * @returns {String} result
 */
export default function ultraShorten(input) {
  let str = input
  let index = 0
  while (index < str.length) {
    for (let i = 0; i < str.length; i += 1) {
      if (str[i] === str[i + 1]) {
        str = str.substring(0, i) + str.substring(i + 2)
        index = 0
        break
      }
      index += 1
    }
  }
  return str || 'Empty String'
}
